src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang

```diff
@@ -0,0 +1,102 @@
+module sonic-bgp-allowed-prefix {
+    namespace "http://github.com/Azure/sonic-bgp-allowed-prefix";
+    prefix bgppre;
+    yang-version 1.1;
+
+    import sonic-bgp-common {
+        prefix bgpcmn;
+    }
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-routing-policy-sets {
+        prefix rpolsets;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC BGP Allowed Prefix";
+
+    revision 2022-02-26 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-bgp-allowed-prefix {
+        container BGP_ALLOWED_PREFIXES {
+            list BGP_ALLOWED_PREFIXES_COM_LIST {
+                key "deployment id community";
+
+                leaf deployment {
+                    type string {
+                        pattern "DEPLOYMENT_ID";
+                    }
+                    description "BGP allowed prefix list key type";
+                }
+
+                leaf id {
+                    type uint32;
+                    description "BGP allowed prefix list deployment id";
+                }
+
+                leaf community {
+                    type string;
+                    description "BGP allowed prefix list deployment community";
+                }
+
+                leaf default_action {
+                    type rpolsets:routing-policy-action-type;
+                    description "Permit/Deny action for BGP allow prefix list";
+                }
```

Comment:
BGP_ALLOWED_PREFIXES_COM_LIST and BGP_ALLOWED_PREFIXES_LIST have the same leaf: default_action, prefixes_v4 and prefixes_v6.
Can you use grouping?

Reply:
@ganglyu good suggestion, but there is an issue when current sonic_yang_ext.py to tackle with leaf-list + grouping case, it failed to parse the yang models. I filed an issue to track this one.
https://github.com/Azure/sonic-buildimage/issues/10116